simulation-scripts/perturb.sh

```diff
@@ -51,68 +38,61 @@ THIS_SCRIPT="${DIR}/$(basename "$0")"
 
 # required defaults
 declare -a ARGUMENTS
-EXPECTED_NUM_ARGUMENTS=1
 ARGUMENTS=()
 SCENARIO=''
 MASTER_HOST=""
 SLAVE_HOSTS=""
 BASTION_HOST=""
-IS_TEST_ONLY=0
-IS_FORCE=0
 
 main() {
 
-  [[ $# = 0 && "${EXPECTED_NUM_ARGUMENTS}" -gt 0 ]] && usage
+  [[ $# = 0 ]] && usage
 
-  parse_arguments "$@"
-  validate_arguments "$@"
+  while [ $# -gt 0 ]; do
+    case $1 in
+      -h | --help) usage ;;
+      -m | --master)
+        shift
+        not_empty_or_usage "${1:-}"
+        MASTER_HOST="${1}"
+        ;;
+      -s | --slaves)
+        shift
+        not_empty_or_usage "${1:-}"
+        SLAVE_HOSTS="${1}"
+        ;;
+      -b | --bastion)
+        shift
+        not_empty_or_usage "${1:-}"
+        BASTION_HOST="${1}"
+        ;;
+      --)
+        shift
+        break
+        ;;
+      -*) usage "${1}: unknown option" ;;
+      *) ARGUMENTS+=("$1") ;;
+    esac
+    shift
+  done
 
-  local SCENARIO_DIR="scenario/${SCENARIO}/"
+  [[ "${#ARGUMENTS[@]}" -gt 1 ]] && error "Only one scenario accepted"
 
-  if [[ "${IS_TEST_ONLY:-}" == 1 ]]; then
-    success "In test mode, skipping deployment of ${SCENARIO}"
-  else
+  [[ ${MASTER_HOST:-} ]] || error "--master must be an IP or hostname, or comma-delimited list"
+  [[ ${SLAVE_HOSTS:-} ]] || error "--slaves must be an IP or hostname, or comma-delimited list"
 
-    source test-func.sh
+  SCENARIO="${ARGUMENTS[0]:-}" || true
 
-    local FOUND_SCENARIO
-    if FOUND_SCENARIO=$(find_scenario); then
-      if [[ "${IS_FORCE}" != 1 && "${FOUND_SCENARIO}" == "${SCENARIO}" ]]; then
-        if ! is_special_scenario; then
-          error "Scenario ${SCENARIO} already deployed, reset deployment with 'cleanup' first"
-        fi
-      fi
-      info "Found scenario ${FOUND_SCENARIO}"
-    fi
+  local SCENARIO_DIR="scenario/${SCENARIO}/"
 
-    info "Running ${SCENARIO_DIR} against ${MASTER_HOST}"
-  fi
+  info "Running ${SCENARIO_DIR} against ${MASTER_HOST}"
 
-  if ! is_master_accessible; then
+  if ! $(ssh -F "${SSH_CONFIG_FILE}" -o "StrictHostKeyChecking=no" -o "UserKnownHostsFile=/dev/null" -o "ConnectTimeout 3" root@"${MASTER_HOST}" true); then
```

Comment:
This change reduces clarity of intention

Reply:
I can see that, I'll have a rethink

Reply:
reverted back